### PR TITLE
fix(shared): get ipv4 interfaces should be uniq & set @types/fs-extra…

### PR DESCRIPTION
**Pull request by southorange0929**
… as devDeps

## Summary

Sometimes, we will get many ipv4 address.But some are same,so address should be uniq.
image

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/shared/src/url.ts [hunk 1]
```diff
@@ -37,7 +38,7 @@ const getIpv4Interfaces = () => {
       }
     });
   });
-  return ipv4Interfaces;
+  return uniq(ipv4Interfaces);
```
Comment (chenjiahan) [inline packages/shared/src/url.ts:41]: Thanks for your contribution! 😄

I'm wondering that if `lodash.uniq` can be used to uniq objects.

I have written some simple test and it does not work:

image

Also Rsbuild will remove `lodash` from dependencies in the future (to reduce dependencies)

So can we use another method to uniq the ipv4Interfaces?

Comment (southorange0929) [reply packages/shared/src/url.ts:41]: Got it,let me see.

Comment (southorange0929) [reply packages/shared/src/url.ts]: I use map to uniq address, i'm not sure map is good choose for us.But i think it's better than object

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
LGTM!

**Pull request merged**